### ISSUE TITLE
Update products does not work

### ISSUE THREAD

**Issue by TomasAxelsson** [XiaoFaye/WooCommerce.NET#68]
Wordpress version: 4.6.1–sv_SE
WooCommerce.NET: 0.5.3

It's working fine for me to add products with pictures, categories etc.

But, when I try to update products it doesn't work at all.

My code below: (the code works fine for adding products).

      WooCommerceNET.RestAPI rest = new WooCommerceNET.RestAPI("http://url.com/wp-json/wc/v3/", "key", "key");
           WooCommerceNET.WooCommerce.Legacy.WCObject wc = new WooCommerceNET.WooCommerce.Legacy.WCObject(rest);

 await wc.UpdateProduct(4302, new Product { name = "test 9" });

Error:

   vid System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   vid System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   vid System.Delegate.DynamicInvokeImpl(Object[] args)
   vid System.Windows.Forms.Control.InvokeMarshaledCallbackDo(ThreadMethodEntry tme)
   vid System.Windows.Forms.Control.InvokeMarshaledCallbackHelper(Object obj)
   vid System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   vid System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   vid System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   vid System.Windows.Forms.Control.InvokeMarshaledCallback(ThreadMethodEntry tme)
   vid System.Windows.Forms.Control.InvokeMarshaledCallbacks()
   vid System.Windows.Forms.Control.WndProc(Message& m)
   vid System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   vid System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   vid System.Windows.Forms.NativeWindow.DebuggableCallback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
   vid System.Windows.Forms.UnsafeNativeMethods.DispatchMessageW(MSG& msg)
   vid System.Windows.Forms.Application.ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(IntPtr dwComponentID, Int32 reason, Int32 pvLoopData)
   vid System.Windows.Forms.Application.ThreadContext.RunMessageLoopInner(Int32 reason, ApplicationContext context)
   vid System.Windows.Forms.Application.ThreadContext.RunMessageLoop(Int32 reason, ApplicationContext context)
   vid System.Windows.Forms.Application.Run(Form mainForm)
   vid wootest.Program.Main() i c:\users\tomas axelsson\documents\visual studio 2015\Projects\wootest\wootest\Program.cs:rad 19
   vid System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
   vid System.AppDomain.ExecuteAssembly(String assemblyFile, Evidence assemblySecurity, String[] args)
   vid Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()
   vid System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   vid System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   vid System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   vid System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   vid System.Threading.ThreadHelper.ThreadStart()

Inner exception: {"Index out of bounds."}

I have double checked the product id exists etc.

Does someone have a clue?

Best regards
Tomas

**Comment by TomasAxelsson** [comment]
I have solved it myself, it was not an issue with WooCommerce.Net it was I who hade not configured my installation of the website correct. So, my mistake sorry for posting.

Anyway, the problem was that I had to tell apache to accecpt PUT and DELETE, so I added the lines you see below in the .htaccess file. So if you are having problems with update products but can create them, this is probably the sollution.

<Limit GET POST PUT DELETE>
  Allow from all
</Limit>
